### PR TITLE
Make hyprctl setcursor better (support XCursor themes, also give fail message)

### PR DESCRIPTION
**Pull request by ikalco**
#### Describe your PR, what does it fix/add?
makes it so that hyprctl setcursor can be used to change cursor theme between Hyprcursor and Xcursor

also addresses these issues
* #4221, by removing this code
  https://github.com/hyprwm/Hyprland/blob/de9798fcf9494eb082bd168175390c0d47b8478b/src/managers/CursorManager.cpp#L269-L270

* #5232, and adding this too
https://github.com/hyprwm/Hyprland/blob/deb7c4f9b07d358d4e01fe8726f72ce9bbc016a2/src/debug/HyprCtl.cpp#L1066-L1069

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
**_This PR also requires a change in hyprcursor, see hyprwm/hyprcursor#43_**

The git blame for these lines is for a bug fix, but I believe this was left over when CursorManager was later changed?
either way code should still work the same, just a little less efficient i guess
https://github.com/hyprwm/Hyprland/blob/de9798fcf9494eb082bd168175390c0d47b8478b/src/managers/CursorManager.cpp#L269-L270

#### Is it ready for merging, or does it need work?
No, I need to get hyprcursor PR merged first (hyprwm/hyprcursor#43)
tested on my machine
* switching between xcursor and hyprcursor 
* and between xcursor and xcursor
* and hyprcursor and hyprcursor
* and different sizes for both

### CODE REVIEW

**Comment by vaxerski** [comment]
shit aint compilin doe

**Comment by ikalco** [comment]
You gotta compile hyprcursor and make Hyprland compile with new header and lib.
I just cmake installed the other PR and then compiled and ran Hyprland

**Comment by vaxerski** [comment]
sorry I skim over long ass descriptions havent caught that, mark it as draft if it depends on something and link the deps

**Comment by ikalco** [comment]
ummm, how do I make build work with hyprcursor-git or jus wait

**Comment by vaxerski** [comment]
remind me later tomorrow if I forget to make a new hc release, then we bump the nix flake and wait for arc lonix

**Comment by ikalco** [comment]
reminder

**Comment by vaxerski** [comment]
thanks! done. @fufexan can you bump the flake here? then we just wait for arc

**Comment by ikalco** [comment]
should be gtg now

File: src/managers/CursorManager.cpp [hunk 1]
```diff
@@ -266,9 +266,6 @@ void CCursorManager::updateTheme() {
             highestScale = m->scale;
     }
 
-    if (std::round(highestScale * m_iSize) == m_sCurrentStyleInfo.size)
```
Comment (vaxerski) [inline src/managers/CursorManager.cpp:269]: why?

File: src/managers/CursorManager.cpp [hunk 2]
```diff
@@ -286,13 +283,52 @@ void CCursorManager::updateTheme() {
...
+
+    if (m_pWLRXCursorMgr)
+        wlr_xcursor_manager_destroy(m_pWLRXCursorMgr);
+
+    m_pWLRXCursorMgr = wlr_xcursor_manager_create(name.empty() ? "" : name.c_str(), size);
+    bool xSuccess    = wlr_xcursor_manager_load(m_pWLRXCursorMgr, 1.0) == 1;
+
+    // this basically checks if xcursor changed used theme to default but better
+    bool                              diffTheme = false;
+    struct wlr_xcursor_manager_theme* theme;
```
Comment (vaxerski) [inline src/managers/CursorManager.cpp:309]: struct is redundant

File: src/managers/CursorManager.cpp (hunk 1 above)
Comment (ikalco) [reply src/managers/CursorManager.cpp:269]: It fixes a bug where you had to apply xcursor themes twice to have it actually be applied
also
>The git blame for these lines is for a bug fix, but I believe this was left over when CursorManager was later changed?
either way code should still work the same, just a little less efficient i guess

Comment (vaxerski) [reply src/managers/CursorManager.cpp:269]: oke right b

File: src/managers/CursorManager.cpp (hunk 2 above)
Comment (ikalco) [reply src/managers/CursorManager.cpp]: sorry was busy a few days
fixde

**Review by vaxerski — APPROVED** [review]
lgtm

**Pull request merged**